encryptonize.go

```diff
@@ -86,76 +90,99 @@ func New(keyProvider key.Provider) (Encryptonize, error) {
 ////////////////////////////////////////////////////////
 
 // Encrypt creates a new sealed object containing the provided plaintext data as well as an access
-// object that controls access to that data. The calling user is automatically added to the access
-// object. To grant other users access, see AddGroupsToAccess and AddUserToGroups.
+// list that controls access to that data. The calling user is automatically added to the access
+// list. To grant other users access, see AddGroupsToAccess and AddUserToGroups.
 //
 // For all practical purposes, the size of the ciphertext in the SealedObject is len(plaintext) + 48
 // bytes.
-//
-// The sealed object and the sealed access object are not sensitive data.
-func (e *Encryptonize) Encrypt(user *data.SealedUser, object *data.Object) (data.SealedObject, data.SealedAccess, error) {
+func (e *Encryptonize) Encrypt(user *data.SealedUser, object *data.Object) (uuid.UUID, error) {
```

Comment:
Shouldn't we also have Encrypt/Decrypt/Update methods that do the encryption in-place and return the encrypted data without using the io provider?

Reply:
I'd be against doing that, unless we have a really good reason, for several reasons:
* It complicates the API
* It complicates the security analysis
* It introduces some weird "bug" scenarios, depending on how we do it (i.e. mixing calls to the two APIs)
* It muddies the story of how encryptonize works
* It's relatively easy to obtain this functionality by using in an in-mem IO provider and wrapping the encryptonize API. Performance might be an issue here, but I don't want to do premature optimization.